R/tools-def.R

```diff
@@ -5,28 +5,75 @@ NULL
 #' Define a tool
 #'
 #' @description
-#' Define an R function for use by a chatbot. The function will always be
-#' run in the current R instance.
+#' Annotate a function for use in tool calls, by providing a name, description,
+#' and type defintion for the arguments.
 #'
 #' Learn more in `vignette("tool-calling")`.
 #'
-#' @param .fun The function to be invoked when the tool is called. The return
+#' # ellmer 0.3.0
+#'
+#' In ellmer 0.3.0, the definition of the `tool()` function changed quite
+#' a bit. To make it easier to update old versions, you can use an LLM with
+#' the following system prompt
+#'
+#' ````
+#' Help the user convert an ellmer 0.2.0 and earlier tool defintion into a
+#' ellmer 0.3.0 tool definition. Here's what changed:
+#'
+#' * All arguments, apart from the first should be named, and the argument
```

Comment:
```suggestion
#' * All arguments apart from the first should be named, and the argument
```

Or add a comma after "first"